lib/gui.py

```diff
@@ -117,60 +134,87 @@ def make_port_scan(main_window):
     port_range = tk.StringVar(value="1-65535")
     threads = tk.StringVar(value=multiprocessing.cpu_count())
 
-    scan_label = tk.Label(port_scan_frame, text="Select scanning options", bg=DEFAULT_BG_COLOR, fg="#FFF")
+    scan_label = tk.Label(
+        port_scan_frame, text="Select scanning options", bg=DEFAULT_BG_COLOR, fg="#FFF"
+    )
     scan_label.pack(fill=tk.X, expand=True)
 
     def validate_input(host_address, ports_range, n_threads):
         host_address = host_address.replace("default: ", "")
         ports_range = ports_range.replace("default: ", "")
         n_threads = n_threads.replace("default: ", "")
         import re
+
         regex = r"^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$"
         ports = ports_range.split("-")
 
         lower_port = ports[0]
         upper_port = ports[1]
         if not re.match(regex, host_address):
             messagebox.showerror("Error", "Invalid host adress")
-        elif len(ports) != 2 or \
-            (lower_port.isdigit() is False or upper_port.isdigit() is False) or \
-            (int(lower_port) < 0 or int(upper_port) < 0) or \
-            (int(upper_port) < int(lower_port)) or \
-                (int(lower_port) > 65535 or int(upper_port) > 65535):
+        elif (
+            len(ports) != 2
+            or (lower_port.isdigit() is False or upper_port.isdigit() is False)
+            or (int(lower_port) < 0 or int(upper_port) < 0)
+            or (int(upper_port) < int(lower_port))
+            or (int(lower_port) > 65535 or int(upper_port) > 65535)
+        ):
             messagebox.showerror("Error", "Invalid port range")
         elif n_threads.isdigit() is False or int(n_threads) < 1:
             messagebox.showerror("Error", "Invalid thread number")
-
         else:
-            result = scan_port_range(host_address, int(lower_port), int(upper_port), int(n_threads))
+            scan_results.delete("1.0", "end-1c")
+            result = scan_port_range(
+                host_address, int(lower_port), int(upper_port), int(n_threads)
+            )
             if len(result) == 0:
-                result_label.config(text="There were not any open ports on specified range")
+                messagebox.showinfo(
+                    "Scanning info", "There were not any open ports on specified range"
+                )
             else:
-                result_label.config(text="Open ports for specified host\n" +
-                                    "\n".join(list(map(lambda x: str(x), result))))
+                scan_results.insert(
+                    tk.END,
+                    "Open ports for specified host\n"
```

Comment:
Let me disagree with using tk.Label. I like tk.Text better because user can manually copy as many port numbers as they want. Label don't allow copying anything using mouse.

Reply:
What. I don't propose we switch to tk.Label. I just want some text *above* the tk.Text describing what the tk.Text below will contain instead of inserting this text into the text box itself

Reply:
Okay. That's a decent idea